chia/data_layer/data_store.py

```diff
@@ -982,14 +983,18 @@ async def set_validated_wallet_generation(self, tree_id: bytes32, generation: in
     async def get_kv_diff(
         self,
         tree_id: bytes32,
-        hash_1: Optional[bytes32],
-        hash_2: Optional[bytes32],
+        hash_1: bytes32,
+        hash_2: bytes32,
         *,
         lock: bool = True,
     ) -> Set[DiffData]:
         async with self.db_wrapper.locked_transaction(lock=lock):
-            old_pairs = set() if hash_1 is None else set(await self.get_keys_values(tree_id, hash_1, lock=False))
-            new_pairs = set() if hash_2 is None else set(await self.get_keys_values(tree_id, hash_2, lock=False))
+            old_pairs = set(await self.get_keys_values(tree_id, hash_1, lock=False))
+            new_pairs = set(await self.get_keys_values(tree_id, hash_2, lock=False))
+            if len(old_pairs) == 0 and hash_1 != bytes32([0] * 32):
+                return set()
+            if len(new_pairs) == 0 and hash_2 != bytes32([0] * 32):
+                return set()
```

Comment:
Is there a test that exercises this logic? If not can we add one?

Reply:
added now